packages/region-config-resolver/src/extensions/index.ts

```diff
@@ -0,0 +1,37 @@
+import { AwsRegionExtensionConfiguration } from "@aws-sdk/types";
+import { Provider } from "@smithy/types";
+
+import { getRealRegion } from "../regionConfig/getRealRegion";
+
+/**
+ * @internal
+ */
+export const getAwsRegionExtensionConfiguration = (runtimeConfig: Partial<{ region: string | Provider<string> }>) => {
+  let runtimeConfigRegion: Provider<string> = async () => {
+    const region = runtimeConfig.region!;
+    if (typeof region === "string") {
+      return region;
+    }
+    return await region();
```

Comment:
don't need return await in async functions. I think it may unwrap and rewrap a Promise value 🤷